internal/external/procfile/generate.go

```diff
@@ -0,0 +1,48 @@
+package procfile
+
+import (
+	"bytes"
+
+	"gopkg.in/alessio/shellescape.v1"
+	"mvdan.cc/sh/v3/syntax"
+)
+
+func Generate(procs []Process) (string, error) {
+	var out bytes.Buffer
+
+	for _, proc := range procs {
+		out.WriteString(proc.Name)
+		out.WriteString(": ")
+
+		cmd := &syntax.CallExpr{
+			Args: []*syntax.Word{
+				literalWord(proc.Program),
+			},
+		}
+		for _, arg := range proc.Arguments {
+			cmd.Args = append(cmd.Args, literalWord(arg))
+		}
+
+		for key, val := range proc.Environment {
+			cmd.Assigns = append(cmd.Assigns, &syntax.Assign{
+				Name:  &syntax.Lit{Value: key},
+				Value: literalWord(val),
+			})
+		}
+
+		syntax.NewPrinter().Print(&out, &syntax.Stmt{
+			Cmd: cmd,
+		})
+		out.WriteByte('\n')
+	}
+
+	return out.String(), nil
+}
+
+func literalWord(val string) *syntax.Word {
+	return &syntax.Word{
+		Parts: []syntax.WordPart{
+			&syntax.Lit{Value: shellescape.Quote(val)},
```

Comment:
Does this work correctly? Is it _expected_ to work correctly? Seems like it might be abusing the fact that the printer doesn't actually inspect the internals of the Lit value... which might be OK? Not sure what else to do about it. Looks like you've got a test though, so works for me!

Reply:
Yeah, it looks like the parser generates `Lit` rather than more specific nodes indicating things like double quoted string, escaped character, etc. The shellescape library generates ugly output, but it is correct.